can/interfaces/socketcand/socketcand.py

```diff
@@ -13,12 +13,96 @@
 import socket
 import time
 import traceback
+import urllib.parse as urlparselib
+import xml.etree.ElementTree as ET
 from collections import deque
+from typing import List
 
 import can
 
 log = logging.getLogger(__name__)
 
+DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS = ""
+DEFAULT_SOCKETCAND_DISCOVERY_PORT = 42000
+
+
+def detect_beacon():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    sock.bind((DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS, DEFAULT_SOCKETCAND_DISCOVERY_PORT))
+    log.info(
+        f"Listening on for socketcand UDP advertisement on {DEFAULT_SOCKETCAND_DISCOVERY_ADDRESS}:{DEFAULT_SOCKETCAND_DISCOVERY_PORT}"
+    )
+
+    # Time between beacons no more than 3 seconds. Allow for at least 3
+    timeout_ms = 12000
```

Comment:
This could be a function parameter.

Reply:
Should this be defined as a global, and use it has a function parameter to detect_beacon()? Or a parameter to _detect_available_configs()? If the later - we would need to add a kwargs to the BusABC::_detect_available_configs(), correct?

Reply:
The timeout should be a function parameter for `detect_beacon` in seconds. `_detect_available_configs` can use a short timeout like 3.1s, while users can wait for longer if they call this function directly.

Reply:
Ok. I made it 3.1s in _detect_available_configs, and added the parameter to detect_beacon.